corehq/motech/models.py

```diff
@@ -153,8 +153,29 @@ def get_requests(
         )
 
     def get_auth_manager(self):
+        # Auth types that don't require a username:
         if self.auth_type is None:
             return AuthManager()
+        if self.auth_type == OAUTH1:
+            return OAuth1Manager(
+                client_id=self.client_id,
+                client_secret=self.plaintext_client_secret,
+                api_endpoints=self._get_oauth1_api_endpoints(),
+                connection_settings=self,
+            )
+        if self.auth_type == OAUTH2_CLIENT:
+            return OAuth2ClientGrantManager(
+                self.url,
+                client_id=self.client_id,
+                client_secret=self.plaintext_client_secret,
+                token_url=self.token_url,
+                refresh_url=self.refresh_url,
+                connection_settings=self,
+            )
+
+        # Auth types that require a username:
+        if not isinstance(self.username, str):
```

Comment:
I considered checking for `None`, or just whether `self.username` is Truthy, but chose explicitly checking it's a `str` because that's what's breaking.